contracts/SpokePoolV3Periphery.sol

```diff
@@ -77,216 +82,99 @@ abstract contract SwapAndBridgeBase is Lockable, MultiCaller {
-     * @dev If swapToken or acrossInputToken are the native token for this chain then this function might fail.
-     * the assumption is that this function will handle only ERC20 tokens.
-     * @param routerCalldata ABI encoded function data to call on router. Should form a swap of swapToken for
-     * enough of acrossInputToken, otherwise this function will revert.
-     * @param swapTokenAmount Amount of swapToken to swap for a minimum amount of depositData.inputToken.
-     * @param minExpectedInputTokenAmount Minimum amount of received depositData.inputToken that we'll submit bridge
-     * deposit with.
-     * @param depositData Specifies the Across deposit params we'll send after the swap.
+     * @notice Passthrough function to `depositV3()` on the SpokePool contract.
+     * @dev Protects the caller from losing their ETH (or other native token) by reverting if the SpokePool address
+     * they intended to call does not exist on this chain. Because this contract can be deployed at the same address
+     * everywhere callers should be protected even if the transaction is submitted to an unintended network.
+     * This contract should only be used for native token deposits, as this problem only exists for native tokens.
+     * @param recipient Address to receive funds at on destination chain.
+     * @param inputToken Token to lock into this contract to initiate deposit.
+     * @param inputAmount Amount of tokens to deposit.
+     * @param outputAmount Amount of tokens to receive on destination chain.
+     * @param destinationChainId Denotes network where user will receive funds from SpokePool by a relayer.
+     * @param quoteTimestamp Timestamp used by relayers to compute this deposit's realizedLPFeePct which is paid
+     * to LP pool on HubPool.
+     * @param message Arbitrary data that can be used to pass additional information to the recipient along with the tokens.
+     * Note: this is intended to be used to pass along instructions for how a contract should use or allocate the tokens.
+     * @param exclusiveRelayer Address of the relayer who has exclusive rights to fill this deposit. Can be set to
+     * 0x0 if no exclusivity period is desired. If so, then must set exclusivityDeadline to 0.
+     * @param exclusivityDeadline Timestamp after which any relayer can fill this deposit. Must set
+     * to 0 if exclusiveRelayer is set to 0x0, and vice versa.
+     * @param fillDeadline Timestamp after which this deposit can no longer be filled.
      */
-    function swapAndBridge(
-        bytes calldata routerCalldata,
-        uint256 swapTokenAmount,
-        uint256 minExpectedInputTokenAmount,
-        DepositData calldata depositData
-    ) external nonReentrant {
-        _swapAndBridge(
-            routerCalldata,
-            swapTokenAmount,
-            minExpectedInputTokenAmount,
-            depositData,
-            SWAP_TOKEN,
-            ACROSS_INPUT_TOKEN
+    function deposit(
```

Comment:
This is taken from the spoke pool verifier. The added changes were to remove the `spokePool` argument, since it should be defined in this contract on initialization, and to add a `nonReentrant` modifier to the function.